File: Codeforces/1490C Sum of Cubes.cpp
#include<bits/stdc++.h>
#define ll long long
using namespace std;
int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0), cout.tie(0);

    ll tc, n, temp, res;
    map<ll, bool>mp;

    for(ll i=1; i<=10005; i++) mp[i*i*i] = 1;
    cin>>tc;
    while(tc--){
        cin>>n;
        res = 0;
        for(ll i=1; i*i*i<n && !res; i++)
            if(mp[i*i*i] && mp[n-i*i*i]) res=1;
        if(!res) cout<<"NO"<<endl;
        else cout<<"YES"<<endl;
    }
    return 0;
}

File: Codeforces/1490C Sum of Cubes.py
tc = int(input())
a = set()
for i in range(1, 10001):
    a.add(i**3)
for cs in range(tc):
    n = int(input())
    res = 0
    for iii in a:
        if n-iii in a:
            res = 1
            break
    if res == 0:
        print("NO")
    else:
        print("YES")
